pvlib/modelchain.py

```diff
@@ -782,19 +756,23 @@ def complete_irradiance(self, times=None, weather=None):
 
         >>> # my_weather containing 'dhi' and 'ghi'.
         >>> mc = ModelChain(my_system, my_location)  # doctest: +SKIP
-        >>> mc.complete_irradiance(my_datetime, my_weather)  # doctest: +SKIP
-        >>> mc.run_model()  # doctest: +SKIP
+        >>> mc.complete_irradiance(my_weather)  # doctest: +SKIP
+        >>> mc.run_model(mc.weather)  # doctest: +SKIP
 
         >>> # my_weather containing 'dhi', 'ghi' and 'dni'.
         >>> mc = ModelChain(my_system, my_location)  # doctest: +SKIP
-        >>> mc.run_model(my_datetime, my_weather)  # doctest: +SKIP
+        >>> mc.run_model(my_weather)  # doctest: +SKIP
         """
-        if weather is not None:
-            self.weather = weather
+        self.weather = weather
+
         if times is not None:
-            self.times = times
+            warnings.warn('times keyword argument is deprecated and will be '
+                          'removed in 0.8. Set the index of the weather '
+                          'argument instead.', pvlibDeprecationWarning)
+
         self.solar_position = self.location.get_solarposition(
-            self.times, method=self.solar_position_method)
+            self.weather.index, method=self.solar_position_method)
```

Comment:
We're not using `times` in any fashion so we're not really backward compatible, just providing a warning message and then moving ahead with new behavior.  More of a comment on the docstring, but if we want to maintain backward compatibility, we should use `times`, and warn if `times` conflicts with `weather.index`.   Same comment on `prepare_inputs`

Reply:
Right, the removal of the deprecated clear sky assumption makes the `times` input superfluous. I'm not opposed to keeping the `times` attribute so long as it's set like `self.times = weather.index`. That's probably the easiest thing to do. We could deprecate the attribute, but I don't think it matters either way. I don't think it makes sense to use `times` to calculate solar position because we'll end up with solar position that's inconsistent with `weather`.